controllers/dspipeline_controller.go

```diff
@@ -194,6 +191,11 @@ func (r *DSPAReconciler) Reconcile(ctx context.Context, req ctrl.Request) (ctrl.
 		return ctrl.Result{}, nil
 	}
 
+	err = params.ExtractParams(ctx, dspa, r.Client, r.Log)
```

Comment:
this is moved after finalizer check because if this fails, it will stall CR deletion, which doesn't make much sense, we should be able to cleanup even if params are invalid for instance.